src/main/java/com/google/api/codegen/viewmodel/StaticLangXExampleView.java

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.api.codegen.SnippetSetRunner;
+import com.google.auto.value.AutoValue;
+
+import java.util.List;
+
+@AutoValue
+public abstract class StaticLangXExampleView implements ViewModel {
+  @Override
+  public abstract String templateFileName();
+
+  @Override
+  public abstract String outputPath();
+
+  @Override
+  public String resourceRoot() {
+    return SnippetSetRunner.SNIPPET_RESOURCE_ROOT;
+  }
+
+  /** The package name of the example */
+  public abstract String exampleLocalPackageName();
+
+  /** The package name of the library */
+  public abstract String libLocalPackageName();
+
+  /** The name of the constructor of the client */
+  public abstract String clientConstructorName();
+
+  /** The name of the example of the constructor of the client */
+  public abstract String clientConstructorExampleName();
+
+  /** Type of the client */
+  public abstract String clientName();
```

Comment:
I didn't add any comments to the existing views because they are largely superfluous. I was planning on writing a central document that would describe all of the concepts involved and point to that from class-level doc of each view model.

If this attribute holds a type, it should be called "clientTypeName".